### PR TITLE
Add specs for exponential math functions.

### PR DESCRIPTION
**Pull request by Awjin**
[skip dart-sass]

### CODE REVIEW

File: spec/core_functions/math/log.hrx [hunk 1]
```diff
@@ -0,0 +1,162 @@
+<===> options.yml
+---
+:todo:
+- sass/libsass#3044
+
+<===>
+================================================================================
+<===> input.scss
```
Comment (nex3) [inline spec/core_functions/math/log.hrx]: A test file shouldn't have subdirectories that also contain tests (this isn't in the style guide but it probably should be), so this should be in its own directory. I'd call it `positive` to match the other specs here.

File: spec/core_functions/math/log.hrx [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+a {b: math.log(2, 0)}
+
+<===> base/zero/output.css
+a {
+  b: 0;
+}
+
+<===>
+================================================================================
+<===> base/one/input.scss
```
Comment (nex3) [inline spec/core_functions/math/log.hrx]: Test a base between zero and one?

File: spec/core_functions/math/log.hrx [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+a {b: math.log(2, 1)}
+
+<===> base/one/output.css
+a {
+  b: Infinity;
+}
+
+<===>
+================================================================================
+<===> error/number_has_units/input.scss
```
Comment (nex3) [inline spec/core_functions/math/log.hrx]: Don't forget to test type errors as well.

File: spec/core_functions/math/log.hrx [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+@use "sass:math" as math;
+a {b: math.log(1px, 1px)}
+
+<===> error/number_and_base_have_units/error
+Error: Expected $number to have no units.
+  ,
+2 | a {b: math.log(1px, 1px)}
+  |       ^^^^^^^^^^^^^^^^^^
+  '
+  input.scss 2:7  root stylesheet
```
Comment (nex3) [inline spec/core_functions/math/log.hrx]: If you wanted to, I think you could omit this.

File: spec/core_functions/math/pow.hrx [hunk 5]
```diff
@@ -0,0 +1,320 @@
+<===> options.yml
+---
+:todo:
+- sass/libsass#3044
+
+<===>
+================================================================================
+<===> input.scss
+@use "sass:math" as math;
+a {b: math.pow(2, 3)}
```
Comment (nex3) [inline spec/core_functions/math/pow.hrx]: It's probably also a good idea to test a real number with a fractional exponent as well.

**Comment by nex3** [comment]
Another thing I forgot—for functions like `log()` that take optional arguments that default to `null`, it's good to add a test for explicitly passing `null` to ensure that some implementation doesn't accidentally make it default to something else (in this case, `$e`).

**Comment by nex3** [comment]
One more thing I forgot (and I may have missed this in the last review, too): all built-in function tests should test passing arguments by name, to ensure that all implementations use the same argument names.

**Comment by Awjin** [comment]
> For functions that have special behavior for certain real numbers, it's probably a good idea to test that they behave that way for numbers that Sass considers equivalent to those numbers (within the precision limit) as well. For example, `math.log(2, 0.000000000001)` should return `-Infinity`, and `math.pow(1.000000000001, Infinity)` should return `NaN`.

Should Sass round between operations? i.e. does `log(log(2))` evaluate to `log(0.30102999566`...`)` or `log(0.3010299957)`?

File: spec/core_functions/math/log.hrx (hunk 1 above)
Comment (Awjin) [reply spec/core_functions/math/log.hrx]: Done.

File: spec/core_functions/math/log.hrx (hunk 2 above)
Comment (Awjin) [reply spec/core_functions/math/log.hrx]: Done.

File: spec/core_functions/math/log.hrx (hunk 3 above)
Comment (Awjin) [reply spec/core_functions/math/log.hrx]: Done.

File: spec/core_functions/math/log.hrx (hunk 4 above)
Comment (Awjin) [reply spec/core_functions/math/log.hrx]: Done.

File: spec/core_functions/math/pow.hrx (hunk 5 above)
Comment (Awjin) [reply spec/core_functions/math/pow.hrx]: Done.
